### PR TITLE
Add disk encryption tutorial

### PR DESCRIPTION
**Pull request by idesai**
Demonstrate an example disk encryption scheme using tpm2-tools.
More recent version of cryptsetup wil have built in support.

Signed-off-by: Imran Desai <imran.desai@intel.com>

### CODE REVIEW

**Comment by idesai** [comment]
@williamcroberts @AndreasFuchsSIT @PeterHuewe please take a look. Thanks!

File: _posts/2020-04-13-Disk-Encryption.md [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+achieves these objectives. This security module is called the Trusted Platform
+Module or TPM. The TPM came about in two major standard specifications, the 1.2
+and the now 2.0. Algorithm flexibility and enhanced authorization schemes are of
+the major advancements in 2.0. Secure persistent storage, platform configuration
+register or PCR for recording system software state(more on this later), a per
+TPM unique certified key for digital identity, and hierarchies for partitioning
+roles based access to TPM objects are of the notable TPM properties. TPM is a
+limited resourced device.
+
+# Intel Platform Trust Technology or Intel(R) PTT:
```
Comment (AndreasFuchsSIT) [inline _posts/2020-04-13-Disk-Encryption.md:28]: I'm not to sure about this section ?
I guess @PeterHuewe would like to add another section on Infineon-TPMs and then someone comes with a section on other TPMs.

We should either make this vendor-independent.

If you really need to place this somewhere, we could maybe add a note at the very end called "Tested on the following platforms" and then Intel and Infineon could add the appropriate statements once that was done. It should then however include the complete machine-spec, but could then also include the tpm-vendor + -version you tested this on.

**Comment by AndreasFuchsSIT** [comment]
One more comment: We should add a reference to the WIP on upstream cryptsetup so people know where to track progress and when to expect native support...

File: _posts/2020-04-13-Disk-Encryption.md (hunk 1 above)
Comment (idesai) [reply _posts/2020-04-13-Disk-Encryption.md]: I will have a separate article on Intel (R) PTT and reference this article in some context there.

**Comment by idesai** [comment]
> One more comment: We should add a reference to the WIP on upstream cryptsetup so people know where to track progress and when to expect native support...
> 
> Besides, I really really like this and the effort put into it.

I added a note on the cryptsetup upstreaming effort.

**Comment by PeterHuewe** [comment]
@idesai thanks for putting this effort in! really appreciated and looks nice.

**Pull request merged**